### PR TITLE
Remove annotations from http.HTTPStatus enum members

### PR DESCRIPTION
**Pull request by justinpawela**
These type annotations are unnecessary and seem to confuse the type system.

See `signal.Signals` and `socket.AddressFamily` for other stdlib `Enum`s (specifically `IntEnum`s) defined without a need for type annotations on each member.

In fact (as of mypy 0.610), the annotations on `http.HTTPStatus` members seems to confuse the type checker if the `--disallow-any-expr` flag is used.

While this produces no mypy warnings:

```python
from signal import Signals

def do_something() -> None:
    status = Signals.SIGABRT
    reveal_type(status)      # Revealed type is 'signal.Signals'
```

This code produces the following warning:

```python
from http import HTTPStatus

def do_something() -> None:
    status = HTTPStatus.OK # Expression type contains "Any" (has type "Type[HTTPStatus]")
    reveal_type(status)    # Revealed type is 'http.HTTPStatus'
```

I'm not sure if this is a mypy bug as well (and could submit a report if needed), but removing the type annotations from members certainly resolves the issue with this enum.

### CODE REVIEW

**Comment by justinpawela** [comment]
I should note that `http.HTTPStatus` seems to be the only `IntEnum` subclass in the stdlib that was annotated this way.

I did however find `plistlib.PlistFormat`, a subclass of plain `Enum`. It suffers from the same per-member type annotations. If core devs agree the annotations should be removed, I can submit another PR. Just let me know.

File: stdlib/3/http/__init__.pyi [hunk 1]
```diff
@@ -10,60 +10,60 @@ if sys.version_info >= (3, 5):
         phrase = ...  # type: str
```
Comment (JelleZijlstra) [inline stdlib/3/http/__init__.pyi]: These will probably also be treated strangely by the type system; can you annotate them as `@property`?

Comment (justinpawela) [reply stdlib/3/http/__init__.pyi]: I don't mind updating `phrase` and `description` as `@property`s, though I should point out that they're not defined that way in the stdlib. They're just instance variables populated during `__new__`. There are no explicit getters for either.

Whatever you think is best; I'm new to mypy and typeshed.

FWIW, haven't (yet) had any type checking problems with either `phase` or `description`.

Comment (JelleZijlstra) [reply stdlib/3/http/__init__.pyi]: OK, let's leave this for now.

**Comment by JelleZijlstra** [comment]
Thanks for your contribution!

**Pull request merged**